File: backend/SmartSusChef.Api/Models/Recipe.cs
namespace SmartSusChef.Api.Models;

public class Recipe
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public ICollection<RecipeIngredient> RecipeIngredients { get; set; } = new List<RecipeIngredient>();
    public ICollection<SalesData> SalesRecords { get; set; } = new List<SalesData>();
}

public class RecipeIngredient
{
    public Guid Id { get; set; }
    public Guid RecipeId { get; set; }
    public Guid IngredientId { get; set; }
    public decimal Quantity { get; set; }

    // Navigation properties
    public Recipe Recipe { get; set; } = null!;
    public Ingredient Ingredient { get; set; } = null!;
}

File: mobile/app/src/main/java/com/smartsuschef/mobile/data/models/Recipe.kt
package com.smartsuschef.mobile.data.models

import java.util.Date

data class Recipe(
    val id: String, // Guid (.NET) --> String in Kotlin
    val storeId: Int,
    val name: String,
    val isSubRecipe: Boolean,
    val isSellable: Boolean,
    val createdAt: String, // DateTime (.NET) -> String (ISO format) in Kotlin
    val updatedAt: String,
    // To connect the two models - Recipe and Ingredient
    val recipeIngredients: List<RecipeIngredient> = emptyList()
)
